sklearn/feature_selection/tests/test_sequential.py

```diff
@@ -137,10 +137,26 @@ def test_pipeline_support():
 
 @pytest.mark.parametrize('n_features_to_select', (2, 3, 4))
 def test_unsupervised_model_fit(n_features_to_select):
+    # Make sure that models without classification labels are not being
+    # validated
+
     X, y = make_blobs(n_features=6)
     sfs = SequentialFeatureSelector(
         KMeans(),
         n_features_to_select=n_features_to_select,
     )
     sfs.fit(X)
     assert(sfs.transform(X).shape[1] == n_features_to_select)
+
+@pytest.mark.parametrize('y', ('no_validation', 1j, 99.9, np.nan, 3))
```

Comment:
I had meant an array of complex numbers sized like X. This certainly tests a lack of validation! Now I'm not sure you need a separate test for None

Reply:
My thinking was to test for any kind of input similar to `y=None`, and yes it seems that we don't need a separate test for None. What would we expect SFS to do in the case of unconventional input with the shape of X (whether complex numbers, NaN, strings, etc)?

Reply:
We expect it to let the underlying estimator deal with it!